clients/drcachesim/tools/func_view.cpp

```diff
@@ -170,21 +171,23 @@ func_view_t::process_memref_for_markers(void *shard_data, const memref_t &memref
         shard->last_was_syscall = memref.marker.marker_value >=
             static_cast<int64_t>(func_trace_t::TRACE_FUNC_ID_SYSCALL_BASE);
     }
-    if (shard->last_was_syscall)
-        return;
     switch (memref.marker.marker_type) {
     case TRACE_MARKER_TYPE_FUNC_ID:
-        if (shard->last_func_id != -1)
+        if (shard->last_func_id != -1 && id2info_.count(shard->last_func_id) > 0)
             shard->prev_noret = id2info_[shard->last_func_id].noret;
         shard->last_func_id = static_cast<int>(memref.marker.marker_value);
         break;
     case TRACE_MARKER_TYPE_FUNC_RETADDR:
         assert(shard->last_func_id != -1);
-        ++shard->func_map[shard->last_func_id].num_calls;
+        // Check that the function is in funclist.log.
+        if (id2info_.count(shard->last_func_id) > 0)
```

Comment:
Added checks to make sure the (syscall) function is in funclist.log.
Even though func_view doesn't really do that for regular functions (https://github.com/DynamoRIO/dynamorio/pull/6907#issuecomment-2287955732)